### PR TITLE
[Scarpet] Add display_title function

### PR DESCRIPTION
**Pull request by altrisi**
_Title_ says it all.

Works with both `StringValue` and `FormattedTextValue`, and supports custom timings.

See docs for more info.

Should I mention in docs that times are per-player and are (AFAIK) reset when player rejoins?

Also that `subtitle` requires to have a running `title` to be shown?

### CODE REVIEW

**Comment by Ghoulboy78** [comment]
I don't see the point, tbh. Why not just add to that pr and add to the print() function? the framework is very much present.

**Comment by altrisi** [comment]
#### About actionbar being there without currently being customizable

It's there because the vanilla title command has it, and that probably means that at some time (in the far future) it will support those. Also it can be done with the same packet.

#### About sending 2 packets for timings (and not resetting to default)

Minecraft requires 2 packets for that (sadly, tested) even though the packet has all the required params, and trying to reset to default would just be guessing, since client could have received a change from a `/title times`, and it will also reset anyway when player rejoins (AFAIK, since it's not saved anywhere).

**Comment by altrisi** [comment]
Because print() would have a lot of arguments if you add times

**Comment by altrisi** [comment]
BTW gnembon could you reopen #546?
It wasn't fixed by that PR, that PR only allowed the fully client-side rules (with `inspectClientSideCommand`, like `fogOff`), and it mentioned that it only worked in carpet-less servers because of this.

**Comment by gnembon** [comment]
Looks good. The only thing I would request is change the name - `send_title` is unclear what is 'sending' - its all packets in the game anyways, for instance `draw_shape` is also sending something to the client to let them draw it. More clear names would be 'display_title', `show_title`, `title_text`, `display_text`. 
`Title` can stay in the name only because the vanillla command that controls it is 'title'

**Comment by altrisi** [comment]
Renamed it and all its references to `display_title()`.

File: src/main/java/carpet/script/api/Auxiliary.java [hunk 1]
```diff
@@ -452,6 +455,81 @@ else if (!interactable && targetBlock == null)
...
+            TitleS2CPacket.Action action;
+            List<Value> targets;
+            Text title = null;
+            int in, out, stay;
+            if (pVal instanceof ListValue)
+            {
+                targets = ((ListValue) pVal).getItems();
+                for (Value player:targets)
+                {
+                    if (!(player instanceof EntityValue && ((EntityValue) pVal).getEntity() instanceof PlayerEntity))
```
Comment (gnembon) [inline src/main/java/carpet/script/api/Auxiliary.java:470]: should be able to accept player names as valid entries as well, You can use 
```
if (pVal instanceof ListValue) targets = pVal.getItems() else targets = SingletonList.of(pVal):
List<ServerPlayerEntity> players = targets.stream().map(v -> EntityValue.getPlayerByValue(.....)).collect() - > that should take care of validation
... proceeed
```

**Comment by gnembon** [comment]
`display_title` sounds better
also please that display_title to editor support files for IDEA and np++

**Comment by gnembon** [comment]
looks good so far. Small details, but should be fine :)

**Comment by altrisi** [comment]
Changes done. Should be ready for review again.

**Review by gnembon — APPROVED** [review]

**Pull request merged**